querybook/server/tasks/disable_scheduled_docs.py

```diff
@@ -1,36 +1,83 @@
+from datetime import datetime, timedelta
+from enum import Enum
+from typing import Optional
 from app.flask_app import celery
 
 from models.schedule import (
     TaskSchedule,
 )
 from app.db import with_session
+from env import QuerybookSettings
+from const.schedule import TaskRunStatus
+from const.impression import ImpressionItemType
+from lib.notify.utils import notify_user
+from lib.scheduled_datadoc.legacy import convert_if_legacy_datadoc_schedule
 from logic.schedule import (
     DATADOC_SCHEDULE_PREFIX,
     update_task_schedule,
     with_task_logging,
+    get_task_run_records,
+    delete_task_schedule,
 )
 from logic.datadoc import get_data_doc_by_id
 from logic.user import get_user_by_id
+from logic.impression import get_viewers_count_by_item_after_date
 from lib.logger import get_logger
 
+
 logger = get_logger(__name__)
 
 
+class DisablePolicy(Enum):
+    INACTIVE_OWNER = "INACTIVE_OWNER"
+    NO_IMPRESSION_FOR_N_DAYS = "NO_IMPRESSION_FOR_N_DAYS"
+    FAILED_FOR_N_RUNS = "FAILED_FOR_N_RUNS"
+
+
+DISABLE_POLICY_TO_REASON = {
+    DisablePolicy.INACTIVE_OWNER.value: "Disabled due to deactivated owner",
+    DisablePolicy.NO_IMPRESSION_FOR_N_DAYS.value: "Disabled due to no one viewing the doc for the past {} days",
+    DisablePolicy.FAILED_FOR_N_RUNS.value: "Disabled due to past {} runs all ended up failing",
+}
```

Comment:
why not just put the reason as the enum value? so that can use `DisablePolicy.INACTIVE_OWNER.name` for the string of `"INACTIVE_OWNER"` if needed and `DisablePolicy.INACTIVE_OWNER.value` for the reason
```
class DisablePolicy(Enum):
     INACTIVE_OWNER = "Disabled due to deactivated owner""
     NO_IMPRESSION_FOR_N_DAYS = "..."
     FAILED_FOR_N_RUNS = "..."
```

Reply:
good idea 👍